eclair-core/src/main/scala/fr/acinq/eclair/router/Validation.scala

```diff
@@ -412,7 +412,7 @@ object Validation {
     } else {
       log.debug("ignoring announcement {} (unknown channel)", u)
       sendDecision(origins, GossipDecision.NoRelatedChannel(u))
-      d
+      d.copy(privateChannels = d.privateChannels - u.shortChannelId)
```

Comment:
I think this change should be applied irregardless of other changes because without removal HCs will be staying in this map until `LocalChannelDown` is received which is somewhat undesired.